Add blog thanking hacktoberfest contribs (#138)
grain-lang/grain-lang.org: src_blog/_posts/Grain-Hearts-Hacktoberfest.md

---
title: Grain 🧡 Hacktoberfest!
subtitle: Check out everything contributed to Grain for Hacktoberfest.
date: 2020-11-13 12:00:00
author: Blaine Bublitz
cover: /blog/photos/cover/wesley-tingey-BCri44lkmhQ-unsplash.jpg
coverAttribution: Wesley Tingey / Unsplash
tags: Hacktoberfest
---

Last month, Grain took part in [DigitalOcean's Hacktoberfest](https://hacktoberfest.digitalocean.com) event, which is a great resource for anyone that wants to get into Open Source. Whether you need guidance on _how_ to contribute or just need ideas for projects to help on, each October they provide a platform (and incentives!) for people to get involved.

We had some amazing developers jump in and contribute to Grain in October! And we wanted to take some time to thank everyone and highlight the work they did.

## Highlighting the Contributors

_Warning:_ Heavy use of exclamation points incoming.

> [@bmakuh](https://github.com/bmakuh) submitted a bunch of Pull Requests for some long-standing improvements that needed to be made, including renaming our output filenames from `.wasm` to `.gr.wasm` and adding support for trailing commas in pretty much every data structure!

Reviewer: ```suggestion
[@bmakuh](https://github.com/bmakuh) submitted a bunch of pull requests for some long-standing improvements that needed to be made, including renaming our output filenames from `.wasm` to `.gr.wasm` and adding support for trailing commas in pretty much every data structure!
```

[@ohana54](https://github.com/ohana54) added many much-needed standard library methods, like `flatMap`/`find`/`findIndex` for Lists and `filter`/`reject` for Maps, while also cleaning up our standard library tests!

> [@fa7ad](https://github.com/fa7ad) cleaned up a massive if-else chain with a constant pattern match (now that we support them) and wrote our `isnt` method to check if two values aren't physically equal!

Reviewer: ```suggestion
[@fa7ad](https://github.com/fa7ad) cleaned up a massive if-else chain with a constant pattern match (now that we support them) and wrote our `isnt` operator to check if two values aren't physically equal!
```

[@lvaniscak](https://github.com/lvaniscak) implemented `find` and `findIndex` methods in our Array standard library to match the List API!

[@clovis1122](https://github.com/clovis1122) wrote `Array.fill` and `Array.fillRange` methods for the standard library!

> [@MScheibel](https://github.com/MScheibel) sent a Pull Request for the `Array.count` method in the standard library... and it was their first Pull Request __ever__!!

Reviewer: ```suggestion
[@MScheibel](https://github.com/MScheibel) sent a pull request for the `Array.count` method in the standard library... and it was their first Pull Request __ever__!!
```

> [@miguelcarvalho13](https://github.com/miguelcarvalho13) showed up and wrote an entire Queue implementation for our standard library. So cool!!

Reviewer: 😉 
```suggestion
[@miguelcarvalho13](https://github.com/miguelcarvalho13) showed up and wrote an entire Queue implementation for our standard library. We're lining up to use it!
```

> [@Dony477](https://github.com/Dony477) added `product` methods for both Lists and Arrays to generate cartesian products of tuples! Again, this was their first Pull Request. Color me impressed!

Reviewer: ```suggestion
[@Dony477](https://github.com/Dony477) added `product` methods for both Lists and Arrays to generate cartesian products of tuples! Again, this was their first pull request. Color me impressed!
```

> [@tmphey](https://github.com/tmphey) contributed `List.take`, `List.takeWhile`, and `List.sub` methods to the standard library, in service of future work on sorting methods. And even contributed the website documentation for these!

Reviewer: ```suggestion
[@tmphey](https://github.com/tmphey) contributed `List.take`, `List.takeWhile`, and `List.sub` methods to the standard library, in service of future work on sorting methods. They even contributed the website documentation for these!
```

> [@ng-marcus](https://github.com/ng-marcus) exploded onto the scene and added __language server protocol__ support to the Grain CLI. And then went on to implement it in our VSCode extension! So we now have inline errors and code lenses in VSCode, which is absolutely mind-blowing! We are also excited to have Marcus as a direct contributor on the [grain-language-server](https://github.com/grain-lang/grain-language-server) project.

Reviewer: ```suggestion
[@ng-marcus](https://github.com/ng-marcus) exploded onto the scene and added __language server protocol__ support to the Grain CLI and then went on to implement it in our VSCode extension! We now have inline errors and code lenses in VSCode, which is absolutely mind-blowing! We are also excited to have Marcus as a direct contributor on the [grain-language-server](https://github.com/grain-lang/grain-language-server) project.
```

## So Many Thanks

> The amount of energy around the Grain project in October was astounding. Even though I was unavailable most of the month, I was watching the Pull Requests roll in while Oscar reviewed and guided everyone to get contributions merged.

Reviewer: ```suggestion
The amount of energy around the Grain project in October was astounding. Even though I was unavailable most of the month, I was watching the pull requests roll in while Oscar reviewed and guided everyone to get contributions merged.
```
